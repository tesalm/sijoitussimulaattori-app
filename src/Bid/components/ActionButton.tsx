import React from 'react';

import { Text, View } from 'react-native';
import { WizardFormColors } from '../../App/colors';
import { t } from '../../assets/i18n';
import Icon from '../../general/icon';
import { actionButtons } from '../styles';

export interface ActionButtonProps {
  action: string;
  active: boolean;
}

export const ActionButton = (props: ActionButtonProps): JSX.Element => {
  return (
    <View style={actionButtons.buttonWithText}>
      <Icon
        iconName={props.action}
        iconHeight={24}
        iconWidth={24}
        tintColor={
          props.active
            ? WizardFormColors.buttonsActive
            : WizardFormColors.buttonsUnactive
        }
      />
      <Text
        style={
          props.active
            ? actionButtons.buttonActive
            : actionButtons.buttonUnactive
        }
      >
        {props.action === 'buy' && t('BidPage.Buy')}
        {props.action === 'sell' && t('BidPage.Sell')}
      </Text>
    </View>
  );
};
