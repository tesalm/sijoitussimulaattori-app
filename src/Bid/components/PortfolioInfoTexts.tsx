import React from 'react';
import { Text, View } from 'react-native';

import { t } from '../../assets/i18n';
import { SinglePortfolio } from '../../PortfolioList/reducer';
import { formatCurrency } from '../../util/general';
import { bidStyles } from '../styles';

export interface PortfolioInfoTextsProps {
  currentStockSymbol?: string;
  portfolioList?: SinglePortfolio[];
  selectedPortfolio?: string;
  stockName: string;
}

// Returns info of the selected portfolio to the bid-screen
export const PortfolioInfoTexts = (
  props: PortfolioInfoTextsProps
): JSX.Element => {
  // Returns selected portfolio as SinglePortfolio.
  function getCurrentPortfolio() {
    if (props.portfolioList) {
      return props.portfolioList.find((portfolio) => {
        return portfolio.name === props.selectedPortfolio;
      });
    }
  }

  // Get the sum of the current stocks in selected portfolio. If there is no stocks, returns 0.
  function getSumOfStockFromPortfolio() {
    const currentPortfolio = getCurrentPortfolio();
    if (
      currentPortfolio &&
      currentPortfolio.portfolioInfo &&
      currentPortfolio.portfolioInfo.portfolio
    ) {
      if (!currentPortfolio.portfolioInfo.portfolio.stocks) {
        return 0;
      } else {
        const portfolioStocks = currentPortfolio.portfolioInfo.portfolio.stocks.find(
          (stock) => {
            return stock.uid === props.currentStockSymbol;
          }
        );
        if (portfolioStocks) {
          return portfolioStocks.amount;
        }
      }
    }
    return 0;
  }

  // TODO: Take away default currency.
  // Returns the total market value of selected portfolio.
  function getPortfolioValue() {
    const currentPortfolio = getCurrentPortfolio();
    if (
      currentPortfolio &&
      currentPortfolio.portfolioInfo &&
      currentPortfolio.portfolioInfo.portfolio
    ) {
      return formatCurrency(
        currentPortfolio.portfolioInfo.portfolio.totalMarketValue,
        '$'
      );
    } else {
      return '0$';
    }
  }

  return (
    <View>
      <Text style={bidStyles.infoText}>
        {t('SumUpPage.YouOwn')}{' '}
        <Text style={bidStyles.infoTextHighlight}>
          {getSumOfStockFromPortfolio()}
        </Text>{' '}
        {t('SumUpPage.StocksOf')} {props.stockName}.
      </Text>
      <Text style={bidStyles.infoText}>
        {t('SumUpPage.YourPortfoliosTotVal')}{' '}
        <Text style={bidStyles.infoTextHighlight}>{getPortfolioValue()}</Text>
      </Text>
    </View>
  );
};
