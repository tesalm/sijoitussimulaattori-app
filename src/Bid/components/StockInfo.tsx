import React from 'react';
import { Text, View } from 'react-native';

import { t } from '../../assets/i18n';
import { getLocale } from '../../util/general';
import { formatRevenue } from '../../util/stock';
import { stockinfo } from '../styles';

export interface StockInfoProps {
  name: string;
  revenue: number;
  updated: Date;
}

export const StockInfo = (props: StockInfoProps): JSX.Element => {
  return (
    <View style={stockinfo.container}>
      <View>
        <Text style={stockinfo.stockText}>{props.name}</Text>
      </View>
      <View style={stockinfo.stockinfoMiddle}>
        <View style={stockinfo.stockinfoMiddleContent}>
          <Text style={stockinfo.valueHeaderSmall}>
            {t('StockPage.RevenueText')}
          </Text>
          <Text style={stockinfo.valueSmall}>
            {formatRevenue(props.revenue)}
          </Text>
        </View>
        <View style={stockinfo.stockinfoMiddleContent}>
          {/* TODO: Here comes the graph */}
        </View>
      </View>
      <View>
        <Text style={stockinfo.updatedText}>
          {t('StockPage.Updated')}: {props.updated.toLocaleString(getLocale())}
        </Text>
      </View>
    </View>
  );
};
