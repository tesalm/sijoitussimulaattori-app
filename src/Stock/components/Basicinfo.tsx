import React from 'react';

import { Text, View, ActivityIndicator } from 'react-native';

import { t } from '../../assets/i18n';
import { stockStyles } from '../styles';

import { Intraday, Metadata } from '../reducers';
import { RootState } from '../../redux/reducers';
import { Dispatch, bindActionCreators } from 'redux';
import { connect } from 'react-redux';

export interface BasicinfoProps {
  metadata?: Metadata;
  intraday?: Intraday;
  metaLoading?: boolean;
  metaError?: Error;
  intraLoading?: boolean;
  intraError?: Error;
  revenue: string;
}

export class Basicinfo extends React.Component<BasicinfoProps> {
  constructor(props: BasicinfoProps) {
    super(props);
  }

  revenueColor = (revenue: number): typeof stockStyles.revenueValueGreen => {
    return revenue >= 0
      ? stockStyles.revenueValueGreen
      : stockStyles.revenueValueRed;
  };

  formatValue = (value: number, currency: string): string => {
    if (currency == 'USD') {
      return value + ' $';
    } else if (currency == 'EUR') {
      return value + ' €';
    }
    return value + ' $';
  };

  render() {
    const {
      metadata,
      intraday,
      metaLoading,
      metaError,
      intraLoading,
      intraError,
      revenue,
    } = this.props;

    if (metaLoading || intraLoading) {
      return (
        <View style={stockStyles.loading}>
          <ActivityIndicator size="large" />
        </View>
      );
    } else {
      if (intraday === undefined || metadata === undefined) {
        let errorMessage;
        if (metaError) {
          errorMessage = metaError.message + ' ';
        }
        if (intraError) {
          errorMessage = errorMessage + intraError.message;
        }
        // TODO: Muokkaa error-teksti käyttäjälle.
        return <Text>Error! {errorMessage} </Text>;
      }
    }

    return (
      <View>
        <View style={stockStyles.basicinfo}>
          <Text style={stockStyles.titleStyle}>
            {metadata.name} ({metadata.symbol})
          </Text>
        </View>
        <View style={stockStyles.basicinfo}>
          <View style={stockStyles.basicinfoSmallerComp}>
            <Text style={stockStyles.valueHeader}>{t('StockPage.Low')}</Text>
            <Text style={stockStyles.value}>
              {this.formatValue(intraday.low, metadata.currency)}
            </Text>
            <Text style={stockStyles.valueHeader}>{t('StockPage.High')}</Text>
            <Text style={stockStyles.value}>
              {this.formatValue(intraday.high, metadata.currency)}
            </Text>
          </View>
          <View style={stockStyles.basicinfoSmallerComp}>
            <Text style={stockStyles.valueHeader}>{t('StockPage.Open')}</Text>
            <Text style={stockStyles.value}>
              {this.formatValue(intraday.open, metadata.currency)}
            </Text>
            <Text style={stockStyles.valueHeader}>{t('StockPage.Close')}</Text>
            <Text style={stockStyles.value}>
              {this.formatValue(intraday.close, metadata.currency)}
            </Text>
          </View>
          <View style={stockStyles.basicinfoSmallerComp} />
          <View style={stockStyles.basicinfoMidComp}>
            <Text style={stockStyles.valueHeaderRightSide}>
              {t('StockPage.Volume')}
            </Text>
            <Text style={stockStyles.valueRightSide}>{intraday.volume}</Text>
            <Text style={stockStyles.valueHeaderRightSide}>
              {t('StockPage.RevenueText')}
            </Text>
            <Text style={this.revenueColor(intraday.open)}>{revenue}</Text>
          </View>
        </View>
        <View style={stockStyles.basicinfo}>
          <View>
            <Text style={stockStyles.valueHeader}>
              {t('StockPage.Updated')}: {intraday.fetchTime.toLocaleString()}
            </Text>
          </View>
        </View>
      </View>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  metadata: state.singleStock.metadata,
  intraday: state.singleStock.intraday,
  metaLoading: state.singleStock.metaLoading,
  metaError: state.singleStock.metaError,
  intraLoading: state.singleStock.intraLoading,
  intraError: state.singleStock.intraError,
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators({}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Basicinfo);
