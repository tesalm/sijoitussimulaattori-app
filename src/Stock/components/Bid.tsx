import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import { stockStyles } from '../styles';
import Icon from '../../general/icon';

import { t } from '../../assets/i18n';
import { NavigationScreenProps } from 'react-navigation';
import { Colors } from '../../App/colors';

interface BidProps {}

type BidPropsWithNavigation = BidProps & NavigationScreenProps;

const Bid = (props: BidPropsWithNavigation): JSX.Element => {
  return (
    <TouchableOpacity onPress={() => props.navigation.navigate('Bid')}>
      <View style={stockStyles.bidView}>
        <View style={stockStyles.bidLogoView}>
          <Icon
            iconName={'transaction'}
            iconHeight={24}
            iconWidth={24}
            tintColor={Colors.baseColor}
          />
          <Text style={stockStyles.bidText}>{t('StockPage.Bid')}</Text>
        </View>
        <Icon
          iconName={'open'}
          iconHeight={24}
          iconWidth={24}
          tintColor={Colors.baseColor}
        />
      </View>
    </TouchableOpacity>
  );
};

export default Bid;
