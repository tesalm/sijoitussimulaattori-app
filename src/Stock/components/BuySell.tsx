import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import { stockStyles } from '../styles';
import Icon from '../../general/icon';

import { t } from '../../assets/i18n';

interface BuySellProps {
  containerName: string;
}

const BuySell = (props: BuySellProps): JSX.Element => {
  return (
    // TODO: Make onPress for these items that go to buy or sell view.
    <TouchableHighlight style={stockStyles.buySellButton}>
      <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
        <Text style={stockStyles.buySellText}>{t(props.containerName)}</Text>
        <Icon iconName={'close.png'} iconHeight={24} iconWidth={24} />
      </View>
    </TouchableHighlight>
  );
};

export default BuySell;
