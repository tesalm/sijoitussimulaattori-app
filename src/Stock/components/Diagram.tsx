import React from 'react';
import { Text, View, ActivityIndicator } from 'react-native';

import { t } from '../../assets/i18n';
import { stockStyles } from '../styles';
import { HistoryData } from '../../MarketScreen/reducers';

interface DiagramProps {
  historyData?: HistoryData;
  historyLoading?: boolean;
  historyError?: Error;
}

const Diagram = (props: DiagramProps): JSX.Element => {
  if (props.historyLoading) {
    return (
      <View style={stockStyles.loading}>
        <ActivityIndicator size="large" />
      </View>
    );
  } else {
    if (props.historyError) {
      // TODO: Muokkaa error-teksti käyttäjälle.
      return <Text>Error! {props.historyError.message}</Text>;
    }
  }

  return (
    <View>
      <Text style={stockStyles.titleStyle}>
        {t('StockPage.RevenueOverYear')}
      </Text>
    </View>
  );
};
export default Diagram;
