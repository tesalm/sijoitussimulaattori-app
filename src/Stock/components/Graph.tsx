import React from 'react';
import { Text } from 'react-native';

import { t } from '../../assets/i18n';
import { HistoryDataQuote } from '../../MarketScreen/reducer';
import { verticalScale } from '../../util/scale';
import { stockContainerStyles } from '../styles';

const { VictoryArea, VictoryChart, VictoryAxis } = require('victory-native');

interface graphProps {
  historydata: Array<HistoryDataQuote>;
}

const Graph = (props: graphProps): JSX.Element => {
  const { historydata } = props;
  if (historydata.length < 2) {
    return <Text>{t('Graph.NoData')}</Text>;
  }
  const maxDate = new Date(historydata[0].date);
  const minDate = new Date(historydata[historydata.length - 1].date);
  const startDate = new Date(
    minDate.getFullYear(),
    minDate.getMonth(),
    minDate.getDate()
  ).getTime();
  const endDate = new Date(
    maxDate.getFullYear(),
    maxDate.getMonth(),
    maxDate.getDate()
  ).getTime();
  const months = ['1','2','3','4','5','6','7','8','9','10','11','12']
          .map((key) => t('Graph.Month.' + key));

  function getDomain(data: Array<HistoryDataQuote>) {
    let min = data[0].close;
    let max = data[0].close;
    for (let i = 1, len = data.length; i < len; i++) {
      let v = data[i].close;
      min = v < min ? v : min;
      max = v > max ? v : max;
    }
    return [min, max];
  }

  return (
    <VictoryChart
      height={verticalScale(180)}
      animate={false}
      padding={stockContainerStyles.graphPadding}
      domainPadding={{ y: verticalScale(20) }}
      scale={{ x: 'time' }}
    >
      {/* x -axis */}
      <VictoryAxis
        invertAxis={true}
        fixLabelOverlap={true}
        tickFormat={(x: string) => {
          if (historydata[historydata.length - 1].date === x) {
            return null;
          }
          const date = new Date(x);
          if (endDate === startDate) {
            let mm = date.getMinutes().toString();
            if (mm.length === 1) {
              mm = '0' + mm;
            }
            return date.getUTCHours() + ':' + mm + '\t';
          }
          if (minDate.getFullYear() !== maxDate.getFullYear()) {
            const yy = date
              .getFullYear()
              .toString()
              .substr(-2);
            return date.getDate() + '. ' + months[date.getMonth()] + ' - ' + yy;
          }
          return date.getDate() + '. ' + months[date.getMonth()] + '\t';
        }}
        style={{
          tickLabels: {
            fontSize: 10,
            fill: '#455A64',
            padding: 2,
            textAnchor: 'end',
          },
          axis: { stroke: 'none' },
        }}
      />

      {/* y -axis */}
      <VictoryAxis
        dependentAxis
        orientation="right"
        tickFormat={(y: number) => {
          if (y >= 1000) {
            return Math.round(y);
          }
          return y;
        }}
        style={{
          tickLabels: {
            fontSize: 10,
            fill: '#455A64',
            padding: 2,
          },
          axis: { stroke: 'none' },
          grid: { stroke: 'gray', opacity: 0.1 },
        }}
      />

      <VictoryArea
        domain={{ y: getDomain(historydata) }}
        interpolation="linear"
        animate={false}
        style={{
          data: {
            fill: '#FF9900', //#00A8EF light blue
            stroke: '#FF8000',
            strokeWidth: 1,
            fillOpacity: 0.3,
          },
        }}
        data={historydata}
        x="date"
        y="close"
      />
    </VictoryChart>
  );
};

export default Graph;
