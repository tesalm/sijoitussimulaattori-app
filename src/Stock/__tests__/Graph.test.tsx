import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';

import { DiagramTest, GraphTestProps } from '../components/Diagram';
import Graph from '../components/Graph';

jest.useFakeTimers();

describe('stockpage graph tests', () => {
  const defaultGraphProps: GraphTestProps = {
    historyData: [],
    intraDay: [],
    historyError: undefined,
  };

  const historyGraphProps: GraphTestProps = {
    historyData: [
      {
        symbol: 'AMZN',
        date: '2019-01-04T00:00:00.000Z',
        open: 66.6,
        high: 66.6,
        low: 66.6,
        close: 107.53,
        volume: 66,
      },
      {
        symbol: 'AMZN',
        date: '2019-01-03T00:00:00.000Z',
        open: 66.6,
        high: 66.6,
        low: 66.6,
        close: 107.6799,
        volume: 66,
      },
      {
        symbol: 'AMZN',
        date: '2019-01-02T00:00:00.000Z',
        open: 66.6,
        high: 66.6,
        low: 66.6,
        close: 107.51,
        volume: 66,
      },
    ],
    intraDay: [
      {
        symbol: 'AMZN',
        date: '2019-01-05T16:00:00.000Z',
        open: 66.6,
        high: 66.6,
        low: 66.6,
        close: 107.53,
        volume: 66,
      },
      {
        symbol: 'AMZN',
        date: '2019-01-05T15:55:00.000Z',
        open: 66.6,
        high: 66.6,
        low: 66.6,
        close: 107.6799,
        volume: 66,
      },
      {
        symbol: 'AMZN',
        date: '2019-01-05T15:50:00.000Z',
        open: 66.6,
        high: 66.6,
        low: 66.6,
        close: 107.51,
        volume: 66,
      },
    ],
    historyError: undefined,
  };

  const errorGraphProps: GraphTestProps = {
    historyData: [],
    intraDay: [],
    historyError: {
      name: 'Network Error',
      message: 'Network connection failed',
    },
  };

  it('renders loading', async () => {
    const component = shallow(<DiagramTest {...defaultGraphProps} />);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with no data', async () => {
    const component = shallow(<DiagramTest {...defaultGraphProps} />);
    jest.runAllTimers(); // Exhaust all tasks queued by setTimeout()
    component.update(); // Force re-render of the component
    expect(component.find(Graph).render()).toMatchSnapshot();
  });

  it('renders correctly with error', async () => {
    const component = renderer
      .create(<DiagramTest {...errorGraphProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with intradata', async () => {
    const component = shallow(<DiagramTest {...historyGraphProps} />);
    jest.runAllTimers();
    component.update();
    expect(component.state()).toMatchSnapshot();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with historydata', async () => {
    const wrapper = mount(<DiagramTest {...historyGraphProps} />);

    jest.runAllTimers();
    wrapper.update();

    wrapper
      .find('ButtonGroup')
      .find(TouchableOpacity)
      .at(4)
      .props().onPress!({} as any);

    jest.runAllTimers();
    wrapper.update();

    expect(wrapper.state()).toMatchSnapshot();
    expect(
      wrapper
        .find(Graph)
        .find('VictoryArea')
        .prop('data')
    ).toMatchSnapshot();
  });
});
