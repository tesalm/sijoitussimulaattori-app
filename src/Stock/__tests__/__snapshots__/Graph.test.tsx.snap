// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stockpage graph tests renders correctly with error 1`] = `
<Text>
  Error! 
  Network connection failed
</Text>
`;

exports[`stockpage graph tests renders correctly with historydata 1`] = `
Object {
  "buttonBarIndex": 4,
  "graphLoading": false,
  "historyInterval": Array [
    Object {
      "close": 107.53,
      "date": "2019-01-04T00:00:00.000Z",
      "high": 66.6,
      "low": 66.6,
      "open": 66.6,
      "symbol": "AMZN",
      "volume": 66,
    },
    Object {
      "close": 107.6799,
      "date": "2019-01-03T00:00:00.000Z",
      "high": 66.6,
      "low": 66.6,
      "open": 66.6,
      "symbol": "AMZN",
      "volume": 66,
    },
    Object {
      "close": 107.51,
      "date": "2019-01-02T00:00:00.000Z",
      "high": 66.6,
      "low": 66.6,
      "open": 66.6,
      "symbol": "AMZN",
      "volume": 66,
    },
  ],
}
`;

exports[`stockpage graph tests renders correctly with historydata 2`] = `
Array [
  Object {
    "close": 107.53,
    "date": "2019-01-04T00:00:00.000Z",
    "high": 66.6,
    "low": 66.6,
    "open": 66.6,
    "symbol": "AMZN",
    "volume": 66,
  },
  Object {
    "close": 107.6799,
    "date": "2019-01-03T00:00:00.000Z",
    "high": 66.6,
    "low": 66.6,
    "open": 66.6,
    "symbol": "AMZN",
    "volume": 66,
  },
  Object {
    "close": 107.51,
    "date": "2019-01-02T00:00:00.000Z",
    "high": 66.6,
    "low": 66.6,
    "open": 66.6,
    "symbol": "AMZN",
    "volume": 66,
  },
]
`;

exports[`stockpage graph tests renders correctly with intradata 1`] = `
Object {
  "buttonBarIndex": 0,
  "graphLoading": false,
  "historyInterval": Array [
    Object {
      "close": 107.53,
      "date": "2019-01-05T16:00:00.000Z",
      "high": 66.6,
      "low": 66.6,
      "open": 66.6,
      "symbol": "AMZN",
      "volume": 66,
    },
    Object {
      "close": 107.6799,
      "date": "2019-01-05T15:55:00.000Z",
      "high": 66.6,
      "low": 66.6,
      "open": 66.6,
      "symbol": "AMZN",
      "volume": 66,
    },
    Object {
      "close": 107.51,
      "date": "2019-01-05T15:50:00.000Z",
      "high": 66.6,
      "low": 66.6,
      "open": 66.6,
      "symbol": "AMZN",
      "volume": 66,
    },
  ],
}
`;

exports[`stockpage graph tests renders correctly with intradata 2`] = `
<View>
  <Text
    style={
      Object {
        "color": "#004D40",
        "fontSize": 33.35,
        "fontWeight": "bold",
        "marginBottom": 31.388235294117646,
      }
    }
  >
    Revenue over a year
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 360,
        "justifyContent": "center",
      }
    }
  >
    <Graph
      historydata={
        Array [
          Object {
            "close": 107.53,
            "date": "2019-01-05T16:00:00.000Z",
            "high": 66.6,
            "low": 66.6,
            "open": 66.6,
            "symbol": "AMZN",
            "volume": 66,
          },
          Object {
            "close": 107.6799,
            "date": "2019-01-05T15:55:00.000Z",
            "high": 66.6,
            "low": 66.6,
            "open": 66.6,
            "symbol": "AMZN",
            "volume": 66,
          },
          Object {
            "close": 107.51,
            "date": "2019-01-05T15:50:00.000Z",
            "high": 66.6,
            "low": 66.6,
            "open": 66.6,
            "symbol": "AMZN",
            "volume": 66,
          },
        ]
      }
    />
  </View>
  <Themed.ButtonGroup
    buttons={
      Array [
        "Today",
        "1 Week",
        "1 Month",
        "3 Months",
        "1 Year",
      ]
    }
    containerStyle={
      Object {
        "height": 60,
        "margin": 0,
      }
    }
    onPress={[Function]}
    selectedButtonStyle={
      Object {
        "backgroundColor": "#F0F0F0",
      }
    }
    selectedIndex={0}
    selectedTextStyle={
      Object {
        "color": "#004D40",
      }
    }
    textStyle={
      Object {
        "fontSize": 20,
      }
    }
  />
</View>
`;

exports[`stockpage graph tests renders correctly with no data 1`] = `
<text>
  No data, choose another timespan
</text>
`;

exports[`stockpage graph tests renders loading 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#004D40",
        "fontSize": 33.35,
        "fontWeight": "bold",
        "marginBottom": 31.388235294117646,
      }
    }
  >
    Revenue over a year
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 360,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="green"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "position": "absolute",
        }
      }
    />
  </View>
  <Themed.ButtonGroup
    buttons={
      Array [
        "Today",
        "1 Week",
        "1 Month",
        "3 Months",
        "1 Year",
      ]
    }
    containerStyle={
      Object {
        "height": 60,
        "margin": 0,
      }
    }
    onPress={[Function]}
    selectedButtonStyle={
      Object {
        "backgroundColor": "#F0F0F0",
      }
    }
    selectedIndex={1}
    selectedTextStyle={
      Object {
        "color": "#004D40",
      }
    }
    textStyle={
      Object {
        "fontSize": 20,
      }
    }
  />
</View>
`;
